app/cmd/start.go

```diff
@@ -146,89 +154,176 @@ func start(c *cli.Context) (err error) {
 		return err
 	}
 
+	var wg sync.WaitGroup
+
+	waitCount := 4
+	if spdkEnabled {
+		waitCount++
+	}
+	wg.Add(waitCount)
+
 	// Start disk server
-	diskGRPCServer, diskGRPCListener, err := setupDiskGRPCServer(diskServiceAddress, spdkServiceAddress, spdkEnabled, shutdownCh)
+	diskGRPCServer, diskGRPCListener, err := setupDiskGRPCServer(diskServiceAddress, spdkServiceAddress, spdkEnabled, shutdown.ch)
 	if err != nil {
 		return err
 	}
 	go func() {
+		defer func() {
```

Comment:
Add `wg.Add(1)` just before the `go func() {`. The concept here is to request when needed, so we don't need to pre-count the number we need for the wait group.

Reply:
Remove WaitGroup. Use errgroup instead.